src/metrics.ts

```diff
@@ -63,8 +65,260 @@ function initializeMetricMaps() {
+          distanceMoved: 0,
+        },
+        'wPe': {
+          initialPosition: 'e2',
+          currentPosition: 'e2',
+          distanceMoved: 0,
+        },
+        'wPf': {
+          initialPosition: 'f2',
+          currentPosition: 'f2',
+          distanceMoved: 0,
+        },
+        'wPg': {
+          initialPosition: 'g2',
+          currentPosition: 'g2',
+          distanceMoved: 0,
+        },
+        'wPh': {
+          initialPosition: 'h2',
+          currentPosition: 'h2',
+          distanceMoved: 0,
+        },
+        'bRa': {
+          initialPosition: 'a8',
+          currentPosition: 'a8',
+          distanceMoved: 0,
+        },
+        'bNb': {
+          initialPosition: 'b8',
+          currentPosition: 'b8',
+          distanceMoved: 0,
+        },
+        'bBc': {
+          initialPosition: 'c8',
+          currentPosition: 'c8',
+          distanceMoved: 0,
+        },
+        'bQd': {
+          initialPosition: 'd8',
+          currentPosition: 'd8',
+          distanceMoved: 0,
+        },
+        'bKe': {
+          initialPosition: 'e8',
+          currentPosition: 'e8',
+          distanceMoved: 0,
+        },
+        'bBf': {
+          initialPosition: 'f8',
+          currentPosition: 'f8',
+          distanceMoved: 0,
+        },
+        'bNg': {
+          initialPosition: 'g8',
+          currentPosition: 'g8',
+          distanceMoved: 0,
+        },
+        'bRh': {
+          initialPosition: 'h8',
+          currentPosition: 'h8',
+          distanceMoved: 0,
+        },
+        'bPa': {
+          initialPosition: 'a7',
+          currentPosition: 'a7',
+          distanceMoved: 0,
+        },
+        'bPb': {
+          initialPosition: 'b7',
+          currentPosition: 'b7',
+          distanceMoved: 0,
+        },
+        'bPc': {
+          initialPosition: 'c7',
+          currentPosition: 'c7',
+          distanceMoved: 0,
+        },
+        'bPd': {
+          initialPosition: 'd7',
+          currentPosition: 'd7',
+          distanceMoved: 0,
+        },
+        'bPe': {
+          initialPosition: 'e7',
+          currentPosition: 'e7',
+          distanceMoved: 0,
+        },
+        'bPf': {
+          initialPosition: 'f7',
+          currentPosition: 'f7',
+          distanceMoved: 0,
+        },
+        'bPg': {
+          initialPosition: 'g7',
+          currentPosition: 'g7',
+          distanceMoved: 0,
+        },
+        'bPh': {
+          initialPosition: 'h7',
+          currentPosition: 'h7',
+          distanceMoved: 0,
+        },
+      };
+
+      // Initialize variables to keep track of the maximum distance and the piece
+      let maxDistance = -1;
+      let maxDistancePiece = null;
+
+      ingestedMoves.forEach(move => {
+        
+        const fromSquare = move.from;
+        const toSquare = move.to;
+        //console.log(`fromSquare: ${fromSquare}`);
+        //console.log(`toSquare: ${toSquare}`);
+
```

Comment:
for code that we're merging in let's clean it up, even if draft will make review easier w/o the commented stuff

Reply:
k sounds good